test/support/rails_7_patch.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+# HACK: to prevent the resetting of instance variables after each request in Rails 7
+# see https://github.com/rails/rails/pull/43735
+if Rails::VERSION::MAJOR >= 7
+  module ActionController
+    module Testing
+      module Functional
+        def clear_instance_variables_between_requests; end
```

Comment:
my only concern is, whether we should go forward and adopt the new rails practice (resetting the instance variables) by rewriting the affected tests in this repo, or should we pull things behind by using this hack

Reply:
One other option is to target the hack at the one controller that causes the issue `ImplicitSerializationTestController` in order to limit the likelihood of instance variables being re-used across requests in future.

ie
```ruby
# frozen_string_literal: true

require 'test_helper'

module ActionController
  module Serialization
    class ImplicitSerializerTest < ActionController::TestCase
      class ImplicitSerializationTestController < ActionController::Base
        include SerializationTesting
        
        ...

        # HACK: to prevent the resetting of instance variables after each request in Rails 7
        # see https://github.com/rails/rails/pull/43735
        def clear_instance_variables_between_requests; end

        def update_and_render_object_with_cache_enabled
          @post.updated_at = Time.zone.now # requires hack above to prevent `NoMethodError: undefined method `updated_at=' for nil:NilClass`

          generate_cached_serializer(@post)
          render json: @post
        end

        ...
```

Reply:
I think that would be a more optimized approach. this way we can apply the hack on the affected controllers exclusively, at the same time influence adopting the new practice before writing new tests.

Also in future iteration, we may proactively work on getting rid of this hack from these controllers by rewriting the tests.